Automattic/wp-calypso
# chore: Migrate `@automattic/magellan-mocha-plugin` tests to Jest

#### Background

In #51779 we forked the package `magellan-mocha-plugin` and incorporated it to our repo.

However, in order to keep the PR small, we didn't migrate the tests from `mocha` (the framework originally used by the package) to `jest` (the main test framework of the repo). This PR refactors those tests.

#### Changes proposed in this Pull Request

* Convert tests to `jest` syntax using https://github.com/skovhus/jest-codemods. Some asserts were converted manually.

* Refactor the code that assumes `mocha` is located in `./packages/magellan-mocha-plugin/node_modules/mocha`. In this case we want these tests to use `./node_modules/mocha`.

* In a similar way, refactor tests that assumed test fixtures are located in `./test_support/`.

* Refactor tests that write to the filesystem to use a random temporary directory, so tests can run in parallel.

* Update to latest `mocha` (otherwise the changes required to make the tests work will be incompatible with the `mocha` version used by e2e tests, specifically when using the deprecated option `--opts`).

* Drop unused dependencies (sinon, chai, eslint).

#### Testing instructions

* Verify unit tests pass.
* Inspect the test logs to ensure the new tests are run.
